File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [0] * len(temperatures)
        n = len(temperatures)
        for i in range(n - 2, -1, -1):
            j = i + 1
            while j < n and temperatures[i] >= temperatures[j]:
                if res[j] == 0:
                    j = n
                    break
                j += res[j]
            
            if j < n:
                res[i] = j - i
        
        return res

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        unordered_map<string, vector<string>> result_map{};

        for (auto &str : strs) {
            vector<int> letter_count(26, 0);

            for (auto &c : str) {
                letter_count[c - 'a'] += 1;
            }

            string key{};

            for (auto &digit : letter_count) {
                key += "#" + to_string(digit);
            }

            result_map[key].push_back(str);
        }

        vector<vector<string>> res{};
        
        for (auto &pair : result_map) {
            res.push_back(pair.second);
        }

        return res;
    }
};
